docs/fundamentals/code-analysis/quality-rules/ca1840.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "CA1840: Use 'Environment.CurrentManagedThreadId'"
+description: "'Environment.CurrentManagedThreadId' is simpler and faster than 'Thread.CurrentThread.ManagedThreadId'."
+ms.date: 11/23/2021
+ms.topic: reference
+f1_keywords:
+  - "CA1840"
+helpviewer_keywords:
+  - "CA1840"
+author: buyaa-n
+dev_langs:
+- CSharp
+- VB
+---
+
+# CA1840: Use `Environment.CurrentManagedThreadId` instead of `Thread.CurrentThread.ManagedThreadId`
+
+| | Value |
+|-|-|
+| **Rule ID** |CA1840|
+| **Category** |[Performance](performance-warnings.md)|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+Using `Thread.CurrentThread.ManagedThreadId` for getting the current managed thread id instead of `Environment.CurrentManagedThreadId`.
```

Comment:
```suggestion
Using `Thread.CurrentThread.ManagedThreadId` for getting the current managed thread ID instead of <xref:System.Environment.CurrentManagedThreadId?displayProperty=fullName>.
```